### PR TITLE
inventory for frontend

### LINKED ISSUE #193: Inventory Frontend
# Description
Show inventory on frontend Page as a item name list

# Tasks
## Tasks backend
List of itemNames in inventory

## Tasks frontend
Show itemNames on the page

### LINKED ISSUE #568: Add the ability to use an item from the inventory
**Is your feature request related to a problem? Please describe.**
We need the ability to use an item from the inventory. 
**Describe the solution you'd like**
The Frontend (already having the list of items from the backend) sends (when an item is clicked) the selected item to the backend and the backend executes  e.g the healing. (Discussed with nils)

**Describe alternatives you've considered**
Handling it in the frontend.

**Additional context**

### PR DESCRIPTION
**Pull request by tamer545**
### What?
Inventory for the Frontend
### Why?
To store and use items
### How?
React
### Testing?
No
### Screenshots (for frontend changes)
No
### Anything Else?
closes #193 
closes #568

### CODE REVIEW

**Comment by corsin-lehre** [comment]
This Feature is complete in the sense of it working, we would need data from the backend to make it actually a feature.

File: fabbwled-frontend/src/components/Inventory.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  const filteredItems = inventory.filter((item) =>
+    item.name.toLowerCase().includes(searchTerm.toLowerCase()),
+  );
+
+  const handleItemClick = (item: Item) => {
+    setSelectedItem(item);
+  };
+
+  const handleActionClick = () => {
+    if (selectedItem) {
```
Comment (corsin-lehre) [inline fabbwled-frontend/src/components/Inventory.tsx:49]: change this to a guard statement
```suggestion
    if (!selectedItem) {
       return;
    }
```

**Review by corsin-lehre — APPROVED** [review]
nice and clean code, one thing could be changed

**Review by MagischerWolf05 — APPROVED** [review]
LGTM

**Review by dbabvwid — APPROVED** [review]

**Review by JJoeeyy — APPROVED** [review]
some good ideas here, good clean code, i like the background

**Review by Desi7 — APPROVED** [review]
lgtm 👍

**Review by oliver-jenny — APPROVED** [review]
Nice clean code, rsolve conflicts

To resolve the package-Lock conflicts you can just merge the develop into you branch, so you have the newest version, delete package-lock and node_modules and then ``npm i`` to generate them again

**Review by elion004 — APPROVED** [review]

**Pull request merged**